File: src/test/java/com/github/xumeng/java/problems/easy/problem0303/NumArray0303Test.java
package com.github.xumeng.java.problems.easy.problem0303;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class NumArray0303Test {

    @Test
    public void test() {
        int[] array = {-2, 0, 3, -5, 2, -1};
        NumArray0303 numArray = new NumArray0303(array);
        assertEquals(1, numArray.sumRange(0, 2));
        assertEquals(-1, numArray.sumRange(2, 5));
        assertEquals(-3, numArray.sumRange(0, 5));
        assertEquals(-2, numArray.sumRange(0, 0));
    }

}

File: src/test/kotlin/com/github/xumeng/kotlin/problems/easy/problem0303/NumArray0303Test.kt
package com.github.xumeng.kotlin.problems.easy.problem0303

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class NumArray0303Test {

    @Test
    fun test() {
        val array: IntArray = intArrayOf(-2, 0, 3, -5, 2, -1)
        val numArray = NumArray0303(array)
        assertEquals(1, numArray.sumRange(0, 2))
        assertEquals(-1, numArray.sumRange(2, 5))
        assertEquals(-3, numArray.sumRange(0, 5))
        assertEquals(-2, numArray.sumRange(0, 0))
    }

}